optaplanner-core/src/main/java/org/optaplanner/core/api/score/constraint/ConstraintMatch.java

```diff
@@ -56,7 +59,7 @@ public String getConstraintName() {
     }
 
     public List<Object> getJustificationList() {
-        return justificationList;
+        return Collections.unmodifiableList(justificationList);
```

Comment:
We're duplicating it in the constructor and wrapping it here in an unmodifiable List on every call?
Can we do that in the constructor then too?

Still not sure if it's worth it though. This costs performance wise.
2 proposals:
Proposal A): in constructor: Collections.unmodifiableList(new ArrayList<>(justificationList))
Proposal B:  in constructor: Collections.unmodifiableList(justificationList)

I suspect that for DRL and CS-B, proposal B is bugfree, so the duplication cost of proposal A is a perf loss.
For CS-D we can duplicate before giving it to that constructor?

Reply:
The copy is there so that it can not be modified from the outside by people having the reference.
The `unmodifiableList` is there, so that the copy can not be modified after the user requests it.
Both are necessary. But I agree that it can be done in a single place.